### DISCUSSION TITLE
Docksal without Docker Desktop

### DISCUSSION THREAD

**Discussion by msankhala** [docksal/docksal#1567] [Ideas]
Docksal has a dependency on Docker Desktop for mac system.  Docksal is free but now Docker desktop is going into a subscription model, should we consider removing dependency on Docker desktop?

As per the Docker Subscription Service Agreement mentioned at https://www.docker.com/products/docker-desktop

> Our Docker Subscription Service Agreement includes a change to the terms for Docker Desktop
--
It remains free for small businesses (fewer than 250 employees AND less than $10 million in annual revenue), personal use, education, and non-commercial open source projects.It requires a paid subscription (Pro, Team or Business), for as little as $5 per user per month, for professional use in larger businesses.The effective date of these terms is August 31, 2021. There is a grace period until January 31, 2022 for those that will require a paid subscription to use Docker Desktop.The Docker Pro, Docker Team, and Docker Business subscriptions now include commercial use of Docker Desktop.Check out our FAQ for more information. Or read our latest blog.

Should Docksal consider the containerd as runtime with lima? https://medium.com/nttlabs/containerd-and-lima-39e0b64d2a59

**Comment by lmakarov** [comment]
If Lima can handle the local development setup on Mac similar to how Docker Desktop OR docker-machine + boot2docker (VirtualBox mode) are handling it, then it can be considered as an alternative to Docker Desktop.

From https://github.com/lima-vm/lima:

> Lima launches Linux virtual machines with automatic file sharing, port forwarding, and containerd.
> Lima can be considered as a some sort of unofficial "macOS subsystem for Linux", or "containerd for Mac".

Some other options to consider:

- "Keep calm and ..." stick with Docker Desktop (unless you are a company with 250+ employees AND $10M+ in annual revenue)
- Fork https://github.com/boot2docker/boot2docker and continue to support the (legacy) VirtualBox mode in Docksal
  - this would be the least disruptive option for Docksal
- Switch to https://github.com/kubernetes/minikube - maybe, need to assess the amount of work involved.
  - This project is backed by the Kubernetes community 💪 
  - https://itnext.io/goodbye-docker-desktop-hello-minikube-3649f2a1c469
  - https://arnon.me/2021/09/replace-docker-with-minikube/
- Rancher Desktop - https://github.com/rancher-sandbox/rancher-desktop
  - no idea how good this is.
- colima - https://github.com/abiosoft/colima
  - Based on Lima.
  - Seems to be a popular replacement for Docker Desktop. Mac only, though.

If anyone has other recommendation, feel free to drop them in here. As long as they can reliably solve the problem of running Docker on Mac with file sharing and port forwarding, they will absolutely be considered.

BTW, see my detailed feedback on [podman](https://github.com/docksal/docksal/discussions/1570#discussioncomment-1348864) as an alternative.

**Reply by syntlyx** [reply]
colima (https://github.com/abiosoft/colima)?

**Reply by lmakarov** [reply]
There is a buzz around colima. It seems to be what many are moving to to replace Docker Desktop on Mac. Ideally, we need something that works universally on both Mac and Windows.

**Reply by cemah-h** [reply]
Is there any movement on formalizing a move to Colima and writing some documentation? I'm having trouble setting it up (especially as compared to DDEV where it's trivially easy to set up).

**Reply by syntlyx** [reply]
`brew install --HEAD colima docker`

```
colima start \
--profile default \
--activate \
--arch aarch64 \
--cpu 8 \
--disk 60 \
--memory 6 \
--mount-inotify \
--mount-type virtiofs \
--ssh-agent \
--vm-type vz
```

`nano +1149 /usr/local/bin/fin`

replace 

```
if is_mac; then
    echo $(defaults read /Applications/Docker.app/Contents/Info.plist CFBundleShortVersionString) 2>/dev/null
fi
```

to

```
if is_mac; then
    if [ -f /Applications/Docker.app/Contents/Info.plist ]; then
        echo $(defaults read /Applications/Docker.app/Contents/Info.plist CFBundleShortVersionString) 2>/dev/null
    else
        echo "4.37.2"
    fi
fi
```

**Reply by froboy** [reply]
I just tested @sy2nyk's method above and it worked verbatim. Thanks!

**Reply by syntlyx** [reply]
@froboy see https://github.com/docksal/docksal/pull/1843
tested on colima and orbstack

**Comment by obriat** [comment]
About Windows, since WSL2 you can just use the standard docker package of you favorite distro (and use portainer.io if you need an IHM)
